src/Server/InterserverIOHTTPHandler.cpp

```diff
@@ -116,6 +117,7 @@ void InterserverIOHTTPHandler::handleRequest(HTTPServerRequest & request, HTTPSe
         if (auto [message, success] = checkAuthentication(request); success)
         {
             processQuery(request, response, used_output);
```

Comment:
What if exception will be thrown from this line?
Then finalize will not be called.

Reply:
It will, there is a catch section

https://github.com/ClickHouse/ClickHouse/blob/40a99b7431e49aba57a30c1b73a200776fba3cc0/src/Server/InterserverIOHTTPHandler.cpp#L114-L153